bridges/evm/contracts/example/DEX.sol

```diff
@@ -1,129 +1,114 @@
 pragma solidity 0.5.14;
 pragma experimental ABIEncoderV2;
 
-import {SafeMath} from "openzeppelin-solidity/contracts/math/SafeMath.sol";
-
-interface Bridge {
-    struct VerifyOracleDataResult {
-        bytes data;
-        bytes32 codeHash;
-        bytes params;
-    }
-    function relayAndVerify(bytes calldata data)
-        external
-        returns (VerifyOracleDataResult memory result);
-}
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { IBridge } from "../IBridge.sol";
 
 contract DEX {
-    using SafeMath for uint256;
-
-    bytes32 public codeHash;
-
-    mapping(address => mapping(bytes => uint256)) _balances;
-
-    Bridge bridge = Bridge(0x3e1F8745E4088443350121075828F119075ef641);
-
-    constructor(bytes32 _codeHash) public {
-        codeHash = _codeHash;
+  using SafeMath for uint256;
+
+  bytes32 public codeHash;
+
+  mapping(address => mapping(bytes => uint256)) _balances;
+
+  IBridge bridge;
+
+  constructor(IBridge _bridge, bytes32 _codeHash) public {
+    bridge = _bridge;
+    codeHash = _codeHash;
+  }
+
+  function strCmp(string memory a, string memory b)
+    public
+    pure
+    returns (bool)
+  {
+    return keccak256(abi.encodePacked(a)) == keccak256(abi.encodePacked(b));
+  }
+
+  function balanceOf(address account, string memory symbol)
+    public
+    view
+    returns (uint256)
+  {
+    bytes memory key;
+    if (strCmp(symbol, "ADA")) {
```

Comment:
This is crazy inefficient. We are doing `keccak` hash of symbol like 10 times. Why not just store the result in a variable?